common/src/main/java/dev/architectury/event/events/common/InteractionEvent.java

```diff
@@ -0,0 +1,160 @@
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockState;
+
+public interface InteractionEvent {
+    /**
+     * @see LeftClickBlock#click(Player, InteractionHand, BlockPos, Direction)
+     */
+    Event<LeftClickBlock> LEFT_CLICK_BLOCK = EventFactory.createEventResult();
+    /**
+     * @see RightClickBlock#click(Player, InteractionHand, BlockPos, Direction)
+     */
+    Event<RightClickBlock> RIGHT_CLICK_BLOCK = EventFactory.createEventResult();
+    /**
+     * @see RightClickItem#click(Player, InteractionHand)
+     */
+    Event<RightClickItem> RIGHT_CLICK_ITEM = EventFactory.createCompoundEventResult();
+    /**
+     * @see ClientLeftClickAir#click(Player, InteractionHand)
+     */
+    Event<ClientLeftClickAir> CLIENT_LEFT_CLICK_AIR = EventFactory.createLoop();
+    /**
+     * @see ClientRightClickAir#click(Player, InteractionHand)
+     */
+    Event<ClientRightClickAir> CLIENT_RIGHT_CLICK_AIR = EventFactory.createLoop();
+    /**
+     * @see InteractEntity#interact(Player, Entity, InteractionHand)
+     */
+    Event<InteractEntity> INTERACT_ENTITY = EventFactory.createEventResult();
+    /**
+     * @see FarmlandTrample#trample(Level, BlockPos, BlockState, float, Entity)
+     */
+    Event<FarmlandTrample> FARMLAND_TRAMPLE = EventFactory.createEventResult();
+    
+    interface RightClickBlock {
+        /**
+         * Invoked whenever a player right clicks a block.
+         * Equivalent to Forge's {@code PlayerInteractEvent.RightClickBlock} event and Fabric's {@code UseBlockCallback}.
+         *
+         * @param player The player right clicking the block.
+         * @param hand   The hand that is used.
+         * @param pos    The position of the block in the level.
+         * @param face   The face of the block clicked.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
+         */
+        EventResult click(Player player, InteractionHand hand, BlockPos pos, Direction face);
+    }
+    
+    interface LeftClickBlock {
+        /**
+         * Invoked whenever a player left clicks a block.
+         * Equivalent to Forge's {@code PlayerInteractEvent.LeftClickBlock} event and Fabric's {@code AttackBlockCallback}.
+         *
+         * @param player The player left clicking the block.
+         * @param hand   The hand that is used.
+         * @param pos    The position of the block in the level. Use {@link Player#getCommandSenderWorld()} to get the level.
+         * @param face   The face of the block clicked.
+         * @return A {@link EventResult} determining the outcome of the event,
+         * the execution of the vanilla click may be cancelled by the result.
```

Comment:
Could be shortened to just
```suggestion
         * the action may be cancelled by the result.
```